### PR TITLE
Allow to pass ssl context and ssl connection socket factory

### PR DESCRIPTION
**Pull request by alenkacz**
Another attempt to make it possible to have ssl configurable from the outside

### CODE REVIEW

File: src/main/java/com/rabbitmq/http/client/Client.java [hunk 1]
```diff
@@ -86,7 +89,22 @@ public Client(String url, String username, String password) throws MalformedURLE
...
+
+  /**
+   * Construct an instance with the provided url and credentials.
+   * @param url the url e.g. "http://localhost:15672/api/".
+   * @param username the user name.
+   * @param password the password
+   * @throws MalformedURLException for a badly formed URL.
+   * @throws URISyntaxException for a badly formed URL.
+   */
+  public Client(URL url, String username, String password, SSLConnectionSocketFactory sslConnectionSocketFactory, SSLContext sslContext) throws MalformedURLException, URISyntaxException {
```
Comment (michaelklishin) [inline src/main/java/com/rabbitmq/http/client/Client.java]: Should we instead have two constructors, one with an `SSLConnectionSocketFactory` and one with an `SSLContext`? I find that to be a more common way of expressing effectively mutually exclusive arguments.

**Comment by michaelklishin** [comment]
Thank you for following through with this. I think this is quite close to what I had in mind.

**Comment by michaelklishin** [comment]
Thank you!

**Comment by alenkacz** [comment]
@michaelklishin done, but I still kept the full constructor as private to not duplicate the logic inside the constructor (all other constructors are calling this one)

By the way - I guess this class might already be calling for some proper builder, because currently it is not possible to create the instance with just URL and SSLcontext for example

**Comment by alenkacz** [comment]
@michaelklishin any idea when you will be able to release new version? I desperately need this feature :)

**Comment by michaelklishin** [comment]
@alenkacz I largely agree. If you feel strongly about this, we'd consider a pull request that introduces a builder-like interface.

**Comment by michaelklishin** [comment]
@artembilan do you think you can do a `1.1.0` release some time next week? Pretty please :)

**Comment by michaelklishin** [comment]
Filed https://github.com/rabbitmq/hop/issues/91.

**Pull request merged**